sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcFileFormat.scala

```diff
@@ -305,6 +305,12 @@ private[orc] class OrcOutputWriter(
       recordWriter.close(Reporter.NULL)
     }
   }
+
+  /**
+   * If `recordWriterInstantiated` is false, the output file is not pretouched.
```

Comment:
Alternatively, can we fix the Hive ORC data source to always write the file? This seems wrong to me. We should at least write one file even if the input query is empty, to record the output schema.

Reply:
You are right, we shall make this writer consistent with all the others